apps/emqx_machine/src/emqx_machine_sup.erl

```diff
@@ -29,10 +29,18 @@ start_link() ->
 
 init([]) ->
     Terminator = child_worker(emqx_machine_terminator, [], transient),
+    ReplicantHealthProbe = child_worker(emqx_machine_replicant_health_probe, [], transient),
+    %% Must start before `post_boot'.
+    Booter = child_worker(emqx_machine_app_booter, [], permanent),
```

Comment:
we can maybe turn `emqx_machine_boot` into a gen_statem or gen_server.
and use the `init/1` callback to boot-up instead of calling `post_boot` then exit.

Reply:
I think there's still a need for `post_boot`:  the first time we boot, we print the current EMQX version, and start autocluster; that happens only once during the node's uptime, even if it joins other clusters multiple times.

Reply:
but the process is only `init`ialized once, right?
i mean do not make it temporary, start it as a gen_xxx and keep it running, so there is no need to introduce a new module and new process.

Reply:
I think keeping it in a separate module helps to keep things clean and separated, since there's a bunch of stuff already in machine_boot.

I'll move the post_init stuff to the booter process then.